master/internal/api_runs.go

```diff
@@ -403,3 +410,96 @@ func (a *apiServer) MoveRuns(
 	}
 	return &apiv1.MoveRunsResponse{Results: results}, nil
 }
+
+func (a *apiServer) KillRuns(ctx context.Context, req *apiv1.KillRunsRequest,
```

Comment:
it is unfortunate that this can't be implemented like
- GET/SEARCH /runs returns something like `related_tasks: [...task_ids]` for each run
- This is just `killTasks(...taskIDs string)` and it kills a list of tasks by id.

I'm getting out of touch with how possible/easy what I'm proposing is in the current system. But this is what I saw as this the long term vision for the modular execution work. KillRun's existence seems opposed.

I know none of this was communicated well, though. And maybe we need to let go this vision in favor of just "getting &^#$ done" now. @ioga do you have opinions.

Reply:
for clarity the tone of my comment is meant to be more of a discussion/convo, not blocking review.

Reply:
I think when building a "backend for frontend" it's common to make composite APIs for the sake of performance (to save a roundtrip) or convenience. so I won't be opposed to this. otherwise we'd have to disassemble all APIs into a combination of their minimal orthogonal building blocks.

Reply:
i think bffs are good, too. i just wonder if we should've built the parts first, ha.